sensor_control/vision/symbols/src/ShapeDetector.cpp

```diff
@@ -0,0 +1,167 @@
+
+
+#include "ShapeDetector.h"
+
+float ShapeDetector::findAngle(cv::Point p1, cv::Point p2, cv::Point p3) {
+  cv::Point v1 = p2 - p1;
+  cv::Point v2 = p3 - p1;
+  float m1 = sqrt(v1.x * v1.x + v1.y * v1.y);
+  float m2 = sqrt(v2.x * v2.x + v2.y * v2.y);
+  float dot = v1.x * v2.x + v1.y * v2.y;
+
+  float thecos = dot / (m1 * m2);
+
+  return acos(thecos) * 180 / 3.1415;
+}
+float ShapeDetector::chisquared(std::vector<float> observed, float expected) {
+  float sum = 0;
+  for (int i = 0; i < observed.size(); i++) {
+    sum += (observed[i] - expected) * (observed[i] - expected);
+  }
+  return sum / expected;
+}
+
+float ShapeDetector::findMean(std::vector<float> observed) {
+  float sum = 0;
+  for (int i = 0; i < observed.size(); i++) {
+    sum += observed[i];
+  }
+  return sum / observed.size();
+}
```

Comment:
This would be a lot more succinct:  `float average = std::accumulate( v.begin(), v.end(), 0.0)/v.size();`
accumulate is defined in <numeric>